src/main/java/org/folio/list/services/ListService.java

```diff
@@ -115,8 +122,7 @@ public ListSummaryResultsDTO getAllLists(Pageable pageable, List<UUID> ids, List
     );
 
     // List database do not store entity type labels. Only entity type ID is available in List database.
-    // Get the corresponding entity type labels from FQM
-    Map<UUID, String> entityTypeIdLabelPair = getEntityTypeLabels(lists.getContent());
+    Map<UUID, String> entityTypeIdLabelPair = getEntityTypeLabelMap(entityTypeSummaryResponse.entityTypes());
```

Comment:
this took the list of lists, grabbed their entity types, and sent them to FQM asking for labels...but we already had the list of labels from the original entity type response. This saves a round trip request, and decreases the complexity.